### PR TITLE
Separating Redshift connector from `Database` library into a new `AWS` library

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

Related to #5777 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by radeusgd** [comment]
This PR also adds a widget to `Database.connect` for choosing the database type.

image
image

We currently use `_` placeholders for missing required arguments to show to the user that more are needed. Ideally, we will want to use nested widgets.

File: build.sbt [hunk 1]
```diff
@@ -2065,9 +2070,36 @@ lazy val `std-database` = project
     Compile / packageBin / artifactPath :=
       `database-polyglot-root` / "std-database.jar",
     libraryDependencies ++= Seq(
+      "org.graalvm.truffle" % "truffle-api"             % graalVersion       % "provided",
```
Comment (JaroslavTulach) [inline build.sbt]: Why does database need `truffle-api`? Probably copied from somewhere, but the only part of the system that needs `truffle-api` is `runtime*`.

**Review by JaroslavTulach — APPROVED** [review]
`build.sbt` changes are OK from my perspective. Just consider removing `truffle-api` dependencies - probably unnecessary.

**Review by GregoryTravis — APPROVED** [review]

File: build.sbt (hunk 1 above)
Comment (radeusgd) [reply build.sbt]: I was using the type `org.graalvm.Value` in some of the methods, and so this was needed for Java compiler to resolve that type.
I think I won't need this anymore so I guess I can remove it.

File: distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Main.enso
Comment (jdunkerley) [inline distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Main.enso:1]: What is this for?

File: distribution/lib/Standard/AWS/0.0.0-dev/src/Main.enso [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import project.Database.Redshift.Redshift_Details.Redshift_Details
+import project.Database.Redshift.Redshift_Details.AWS_Credential
+
+export project.Database.Redshift.Redshift_Details.Redshift_Details
+export project.Database.Redshift.Redshift_Details.AWS_Credential
+
+from project.Database.Redshift.Redshift_Details.Redshift_Details export Redshift
```
Comment (jdunkerley) [inline distribution/lib/Standard/AWS/0.0.0-dev/src/Main.enso:7]: Let's remove this - I think we might remove the others in due course.

File: engine/runtime/src/main/scala/org/enso/compiler/phase/ImportResolver.scala [hunk 3]
```diff
@@ -246,6 +246,12 @@ class ImportResolver(compiler: Compiler) {
               case Some(tp) =>
                 (imp, Some(BindingsMap.ResolvedImport(imp, exp, tp)))
               case None =>
+                if (
+                  impName.matches("Standard.AWS.Database.Redshift.[a-zA-Z]+")
+                ) {
+                  System.out.println("A")
+                  System.err.println("B")
+                }
```
Comment (jdunkerley) [inline engine/runtime/src/main/scala/org/enso/compiler/phase/ImportResolver.scala]: This looks like debug stuff.

**Review by jdunkerley — APPROVED** [review]

File: distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Main.enso
Comment (radeusgd) [reply distribution/lib/Standard/AWS/0.0.0-dev/src/Database/Main.enso:1]: It was debug, I forgot to clean this up, will do.

File: distribution/lib/Standard/AWS/0.0.0-dev/src/Main.enso (hunk 2 above)
Comment (radeusgd) [reply distribution/lib/Standard/AWS/0.0.0-dev/src/Main.enso:7]: We agreed on Discord to tackle this later.

**Pull request merged**